awsx/ec2/subnetDistributor.test.ts

```diff
@@ -12,9 +12,109 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { SubnetTypeInputs } from "../schema-types";
+import fc from "fast-check";
+import { SubnetSpecInputs, SubnetTypeInputs } from "../schema-types";
 import { getSubnetSpecs, SubnetSpec } from "./subnetDistributor";
 
+function cidrMask(args?: { min?: number; max?: number }): fc.Arbitrary<number> {
+  return fc.integer({ min: args?.min ?? 16, max: args?.max ?? 27 });
+}
+
+function subnetSpecNoMask() {
+  return fc.constantFrom("Private", "Public", "Isolated").map(
+    (type): SubnetSpecInputs => ({
+      type: type as SubnetTypeInputs,
+    }),
+  );
+}
+
+describe("default subnet layout", () => {
+  it("should have smaller subnets than the vpc", () => {
+    fc.assert(
+      fc.property(
+        fc.record({
+          vpcCidrMask: cidrMask(),
+          azs: fc.array(fc.string({ size: "xsmall" }), { minLength: 2, maxLength: 5 }),
+          subnetSpecs: fc.array(subnetSpecNoMask(), { minLength: 1, maxLength: 5 }),
```

Comment:
Isn't 5 azs and 5 subnets too many for a 27 mask?
5 azs is 3 bits, 5 subnets is 3 more, 6+27 = 33 > 32.
Shouldn't that fail?

Reply:
Yes - it does ... the commit is in the next PR. I'll move it into here so this PR is compete. These tests are run with a random seed so the cases it actually tests each time is different. It tests 100 cases by default.